cafe/models.py

```diff
@@ -0,0 +1,236 @@
+
+    'additive' - an additional amount to the barista’s salary if the cafe’s
+    daly income is higher than the minimum rate.
+
+    Each barista have only one rate for each cafe.
+    """
+
+    min_wage = models.IntegerField(blank=False)
+    percent = models.IntegerField(blank=False)
+    additive = models.IntegerField(blank=False)
+    cafe = models.ForeignKey(
+        Cafe, on_delete=models.CASCADE, blank=False, related_name="rates"
+    )
+    barista = models.ForeignKey(
+        Barista, on_delete=models.CASCADE, blank=False, related_name="rates"
+    )
+
+    def __str__(self):
+        return f"Barista: {self.barista} | Cafe: {self.cafe}"
+
+    class Meta:
+        unique_together = ("barista", "cafe")
+        ordering = ["barista", "cafe"]
+
+
+class Shift(models.Model):
+    """
+    The Shift model is responsible for creating a shift for a certain
+    date in a certain cafe for one barista.
+    The shift has two main parameters:
+
+    'date' - shift date.
+
+    'salary' - This is the barista's salary for the current shift. By default,
+    when creating a shift, this field remains empty. Information in this field
+    appears after income for the current date is entered into the Income model.
+
+    One barista can only have one shift in one cafe on a certain date.
+    This checks the clean() method.
+    """
+
+    date = models.DateField(blank=False)
+    salary = models.IntegerField(blank=True, null=True)
+    cafe = models.ForeignKey(
+        Cafe, on_delete=models.CASCADE, blank=False, related_name="shifts"
+    )
+    barista = models.ForeignKey(
+        Barista, on_delete=models.CASCADE, blank=False, related_name="shifts"
+    )
+
+    @staticmethod
+    def validate_rate(cafe, barista):
+        try:
+            Rate.objects.get(cafe=cafe, barista=barista)
+        except Rate.DoesNotExist:
+            raise ValidationError(f"Barista {barista} "
+                                  f"has no rate for '{cafe}' cafe!")
+
+    @staticmethod
+    def validate_shift(date, barista):
+        shift = Shift.objects.filter(date=date, barista_id=barista)
+        if len(shift) != 0:
+            raise ValidationError(
+                {
+                    "barista": f"On {date}, barista {barista}"
+                    f" is already busy in the another cafe."
+                }
+            )
+
+    @staticmethod
+    def income_calculation(shift, date, cafe):
+        rate = Rate.objects.get(cafe=cafe, barista=shift.barista.id)
+        try:
+            income = Income.objects.get(date=date, cafe=cafe)
+            rate_dict = {
+                "min_wage": rate.min_wage,
+                "percent": rate.percent,
+                "additive": rate.additive
+            }
+            shift.salary = calculation_salary(
+                income=income.income,
+                rate=rate_dict
+            )
+        except Income.DoesNotExist:
+            print("No INCOME for this date", Income.DoesNotExist)
```

Comment:
In pet project you can leave prints, but for production it's better to use logging module
See 
- https://docs.djangoproject.com/en/5.0/topics/logging/
- https://medium.com/@torkashvand/a-comprehensive-guide-to-logging-in-django-e041f311bcb7